### ISSUE TITLE
Edge-Case, problem with refreshing from localStorage if key changes

### ISSUE THREAD

**Issue by bkniffler** [astoilkov/use-local-storage-state#17]
Example, I read the users active tenantId from localStorage. The localStorage key contains the users ID, since if another user signs in on the same machine, they will probably have a different active tenantId. 

```
const [tenantId, setTenantId] = useLocalStorageState<string>(
    `${user?.id}_tid`
);
```

If the user changes without the whole component being reloaded (unloaded and reloaded) OR if the user is undefined because he's still being loaded, it seems like useLocalStorageState will not fetch the localStorage value again for the changed key. tenantId will always turn out to be undefined (in my case, because on first render, the user is undefined). 

As a fix, I'm showing a loader while the user is undefined, and I'll use useLocalStorageState only if there is a user. This is probably more clean anyway. But I was surprised that useLocalStorageState didn't work as I expected. 

What do you think?

ps. Great library, keep it up!

**Comment by astoilkov** [comment]
Yes, this is a bug in the library. I will fix it soon. Thanks for the report and for the appreciation!

**Comment by astoilkov** [comment]
This is now fixed in the newest release – 6.0.2. Thanks a lot for the report!
